.github/workflows/update-ic.yml

```diff
@@ -16,22 +16,35 @@ jobs:
       - name: Check new ic version
         id: update
         run: |
-          current_ic_commit=$(cat ./src/canister_tests/Cargo.toml | sed -n 's/.*{\s*git\s*=\s*"https:\/\/github\.com\/dfinity\/ic",\s*rev\s*=\s*"\([a-z0-9]*\)" }.*/\1/p' | head -n 1)
-          echo "current ic commit '$current_ic_commit'"
-          latest_ic_commit=$(curl --silent -H 'Accept: application/vnd.github.v3+json' https://api.github.com/repos/dfinity/ic/commits/master | jq -cMr .sha)
-          echo "latest ic commit '$latest_ic_commit'"
-
-          if [ "$current_ic_commit" != "$latest_ic_commit" ]
-          then
+          shopt -s globstar
+          
+          need_update=false
+          for filename in **/Cargo.toml; do
+            echo Checking "$filename"
+            [ -e "$filename" ] || continue
+            if ! grep -q https://github\.com/dfinity/ic "$filename"; then
+              echo "$filename" does not contain ic dependecy, skipping
+              continue
+            fi
+          
+            current_ic_commit=$(cat "$filename" | sed -n 's/.*{\s*git\s*=\s*"https:\/\/github\.com\/dfinity\/ic",\s*rev\s*=\s*"\([a-z0-9]*\)" }.*/\1/p' | head -n 1)
+            echo "current ic commit '$current_ic_commit'"
+            latest_ic_commit=$(curl --silent -H 'Accept: application/vnd.github.v3+json' https://api.github.com/repos/dfinity/ic/commits/master | jq -cMr .sha)
```

Comment:
what happens if a new commit is pushed to the ic repo as we're iterating?

Reply:
Ah, good point. I'll move this out of the loop.